frontend/src/redux/search/api.js

```diff
@@ -12,4 +12,36 @@ export const search = (params, setData, setLoading) => {
       setLoading(false)
     })
   };
+};
+
+export const recommendations = (recType, setRecommendations) => {
+
+  var dispatchUrl;
+  switch(recType) {
+    case 'tag':
+      dispatchUrl = "/autoComplete/getTags";
+      break;
+    case 'uni':
+      dispatchUrl = "/autoComplete/getUniversities";
+      break;
+    case 'dep':
+      dispatchUrl = "/autoComplete/getDepartments";
+      break;
+    default:
+      return [];
+  }
+
+  return (dispatch) => {
+    dispatch(actions.recommendationsAction());
+    console.log('x');
+
+    api({sendToken: true})
+    .get(dispatchUrl)
+    .then((response) => {
+      setRecommendations(response.data.result);
+      console.log('c');
+      console.log(response.data.result);
+      dispatch(actions.recommendationsCompletedAction());
```

Comment:
console logs need to go. Again, we should set the redux state not the component's state. This logic should warn you in the browser console as well, with something like "you can't change the state of an unmounted component".